core/controllers/base_test.py

```diff
@@ -697,3 +697,57 @@ def test_controller_class_names(self):
                                         msg=error_message)
 
         self.assertGreater(num_handlers_checked, 150)
+
+
+class SignUpTests(test_utils.GenericTestBase):
+
+    def test_error_is_raised_on_opening_new_tab_during_signup(self):
+        """ Test that error is raised if user opens a new tab
+        during signup.
+        """
+        self.login('abc@example.com')
+        response = self.get_html_response(feconf.SIGNUP_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        os.environ['USER_ID'] = ''
+
+        response = self.testapp.post(
+            feconf.SIGNUP_DATA_URL, params={
+                'csrf_token': csrf_token,
+                'payload': json.dumps({
+                    'username': 'abc',
+                    'agreed_to_terms': True
+                })
+            }, expect_errors=True,
+            )
```

Comment:
Deindent by 4 to match opening paren on line 714. Ditto below.

Reply:
Done.